### DISCUSSION TITLE
Janet 1.21.0

### DISCUSSION THREAD

**Discussion by bakpakin** [janet-lang/janet#938] [General]
- C functions `janet_dobytes` and `janet_dostring` will now enter the event loop if it is enabled.
- Fix hashing regression - hash of negative 0 must be the same as positive 0 since they are equal.
- The `flycheck` function no longer pollutes the module/cache
- Fix quasiquote bug in compiler
- Disallow use of `cancel` and `resume` on fibers scheduled or created with `ev/go`, as well as the root
  fiber.

<hr /><em>This discussion was created from the release <a href='https://github.com/janet-lang/janet/releases/tag/v1.21.0'>Janet 1.21.0</a>.</em>

**Comment by tech189** [comment]
The Windows release is misnamed - it should be janet-1.**21**.1-windows-x64-installer.msi not janet-1.**20**.1-windows-x64-installer.msi

**Reply by sogaiu** [reply]
Nice catch.

I tried building janet from source in a Windows environment using 9d0da74347db1b5e601e5977ccd91391ecca0a7f and the resulting msi had `1.20.1` in its name.

I tried tweaking some things around here: https://github.com/janet-lang/janet/blob/9d0da74347db1b5e601e5977ccd91391ecca0a7f/src/conf/janetconf.h#L7-L10 and rebuilt.  That seemed to produce a more appropriate file name.

FWIW, I think the desired name should have `1.21.0` in it, i.e. neither `1.21.1` nor `1.20.1`.

**Reply by tech189** [reply]
Yep you're right, it should be 1.21.0. I noticed as Scoop was failing to update to the latest version because the download to `janet-1.21.0-windows-x64-installer.msi` was a 404. Maybe you could you make a PR for that file?

**Reply by sogaiu** [reply]
I thought about making a PR, but I'm a bit unsure how bakpakin would want to handle this situation.

I looked at past cases preparing for a release and it looks to me like the typical flow involves removing "-dev" to make a release, make the release, and then adding "-dev" back or something.

The current situation is one where a release has already been made and it seems possible that multiple commits would get made in succession so not sure what an appropriate sequence is.

Perhaps @bakpakin will notice this discussion soon enough.

**Reply by bakpakin** [reply]
Released a patch version to fix this - I currently bump the versions manually so I made a mistake here, thanks for finding.
